indico/web/client/js/react/components/Checkbox.jsx

```diff
@@ -0,0 +1,34 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import './Checkbox.module.scss';
+
+export default function Checkbox({label, onChange, ...inputProps}) {
```

Comment:
One thing that's currently missing is support for the [`indeterminate` state](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes).

We don't use it with the FinalCheckbox but we have a few places where we use it with "standalone" checkboxes, for "select all/none" on top of tables (in that case the checkboxes typically don't have a visible label btw).

![image](https://github.com/user-attachments/assets/3c9e251a-8609-4bb8-9d7e-9a6b089e1712)

Reply:
Ok, I'll add that.

I've also noticed this:

<img width="303" alt="image" src="https://github.com/user-attachments/assets/d291e673-b725-4271-8b2c-0c6d62f02d5a">

I feel that adding non-standard props would not be amazing for this, but I don't have a better idea short of creating a separate component (`ToggleButton` or something), which also doesn't sound amazing. 🤣

Reply:
I've added support for the `indeterminate` state. It is passed in as a Boolean prop, separately from the `checked` prop, like you would with standard checkboxes. Separately, I've added a `showAsToggle` prop which makes it look like a toggle button (it's just cosmetics).